Move section-creation-from-filename, to Section model

marreman opened the pull request:
@idlefingers :boar:

Commit:
Move section-creation-from-filename, to Section model

marreman commented on spec/components/section_spec.rb:
```diff
@@ -8,6 +8,24 @@ class ExampleSection < Ninetails::Section
+    it "should handle absolute paths" do
+      expect(Ninetails::Section.new_from_filename("/foo/bar/billboard.rb")).to be_a Section::Billboard
+    end
+  end
```
Come to think about it.. We are essentially testing `File.basename`.. :P

idlefingers replied:
Yeah, kinda, but it's testing that it instantiates the correct `Section`, so I'm happy..

Commit:
Split new_from_filename into two methods

One that handles converting a lowercase name to an instance of the
class. Another that converts a full filename and then uses the
formentioned method.
